### ISSUE TITLE
jinja[invalid] incorrectly fires when using ansible.builtin.mandatory filter with a msg

### ISSUE THREAD

**Issue by nervo** [ansible/ansible-lint#3908]
##### Summary

ansible-lint reports a `jinja[invalid]` violation when using an `ansible.builtin.mandatory` filter with a msg. Please note that the exact msg value is used as the violation detail, such as:
```
jinja[invalid]: My message
```

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console
ansible-lint 6.22.0 using ansible-core:2.16.0 ansible-compat:4.1.10 ruamel-yaml:0.18.5 ruamel-yaml-clib:0.2.8
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

playbook `test.yaml`:
```yaml
---

- name: Test
  hosts: all
  tasks:
    - name: Test 1
      ansible.builtin.debug:
        msg: "{{ foo | ansible.builtin.mandatory }}"
    - name: Test 2
      ansible.builtin.debug:
        msg: "{{ foo | ansible.builtin.mandatory(msg='My message') }}"

```

```console
ansible-lint test.yaml
```

##### Desired Behavior

Expected no linting violations

```console
Passed: 0 failure(s), 0 warning(s) on 1 files. Profile 'production' was required, and it passed.
```

##### Actual Behavior

A linting violation where `ansible.builtin.mandatory` is used with a `msg` parameter

```console
WARNING  Listing 1 violation(s) that are fatal
jinja[invalid]: My message
test.yaml:11 Task/Handler: Test 2

Read documentation for instructions on how to ignore specific rule violations.

              Rule Violation Summary               
 count tag            profile rule associated tags 
     1 jinja[invalid] basic   formatting           

Failed: 1 failure(s), 0 warning(s) on 1 files. Profile 'production' was required, but 'min' profile passed.
```

Thanks for the hard work, guys :) maintaining such a linter  might be really hard...
